openeo/extra/job_management.py

```diff
@@ -886,3 +936,191 @@ def create_job_db(path: Union[str, Path], df: pd.DataFrame, *, on_exists: str =
     else:
         raise NotImplementedError(f"Initialization of {type(job_db)} is not supported.")
     return job_db
+
+
+class UDPJobFactory:
+    """
+    Batch job factory based on a parameterized process definition
+    (e.g a user-defined process (UDP) or a remote process definition),
+    to be used together with :py:class:`MultiBackendJobManager`.
+
+    Usage example with a remote process definition:
+
+    .. code-block:: python
+
+        from openeo.extra.job_management import (
+            MultiBackendJobManager,
+            create_job_db,
+            UDPJobFactory,
+        )
+
+        # Job creator, based on a parameterized openEO process definition
+        job_starter = UDPJobFactory(
+            namespace="https://example.com/my_process.json",
+        )
+
+        # Initialize job database from dataframe, with parameters to use.
+        df = pd.DataFrame(...)
+        job_db = create_job_db("jobs.csv").initialize_from_df(df)
+
+        # Create and run job manager
+        job_manager = MultiBackendJobManager(...)
+        job_manager.run_jobs(job_db=job_db, start_job=job_starter)
+
+    The factory will take care of filling in the process parameters
+    based on matching column names in dataframe from the job database,
+    with some additional override/fallback options:
+
+    -   When provided, ``parameter_column_map`` will be consulted
```

Comment:
we would need to release some use case example on how to create such a dataframe. It is well documented, but for me not instantly clear what I should fill in in the parameter column map column

Reply:
the `parameter_column_map`  is only for advanced use cases and most users won't have to use it I think. I'll update the docs to make this a bit more clear

Reply:
should be more clear now